.gitattributes

```diff
@@ -0,0 +1,6 @@
+* text=auto
+
+# Linux-centric files maintain LF line endings on Windows platforms
+*.sh        eol=lf
+Makefile*   eol=lf
+configure   eol=lf
```

Comment:
I think it would be helpful to make this a separate patch.
The smaller a patch, the easier it is to accept.

And this in particular is a separate concept from this pull request's title.

That said, I think this is a good idea to enforce text files use lf line endings. It would be unfortunate for someone to submit a pull request where their editor auto-converted opened files. But * text=auto is a tad dangerous. Maybe instead list out the file types we expect? I would include all code files.

Reply:
@ProgramMax How is it dangerous? I recommend reading the [gitattributes documentation page](https://git-scm.com/docs/gitattributes) for more information on what this does.

Other great reading material:

* https://stackoverflow.com/a/10855862/157971

Separate patch is a good idea, I will do that later. However it is likely that I close this PR and move it to the zlib-ng repository per a previous comment. This one is far too inactive and turnaround times for PRs are very long.

Reply:
To your point, even if git assumes wrong for a file type, I think we should explicitly specify only the types it is getting wrong. This is a little more maintainable, as I feel confident the majority of file types will be handled correctly. So in your case we'd do something like:

```
* text=auto
*.png binary
```

The idea here is that the exclusive list will be much larger than the inclusive one. For the most part I trust Git, and as I've seen this work in many projects over many years with very few edge cases coming up, I plan to continue following this pattern in the `.gitattributes` I am contributing. I do appreciate your feedback though, however I think those cases should be handled explicitly.

On that note, does this repo even have PNG files?

Reply:
Also I stumbled upon [this SO question][1], where they state that the issue you're talking about (using PNG as an example) was [fixed in 2.10.0 of Git][2].

[1]: https://stackoverflow.com/questions/29435156/what-will-text-auto-eol-lf-in-gitattributes-do
[2]: https://github.com/git/git/blob/master/Documentation/RelNotes/2.10.0.txt#L248